util/conditions/v1beta2/summary.go

```diff
@@ -0,0 +1,134 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta2
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+// SummaryOption is some configuration that modifies options for a summary call.
+type SummaryOption interface {
+	// ApplyToSummary applies this configuration to the given summary options.
+	ApplyToSummary(*SummaryOptions)
+}
+
+// SummaryOptions allows to set options for the summary operation.
+type SummaryOptions struct {
+	mergeStrategy                  MergeStrategy
+	conditionTypes                 []string
+	negativePolarityConditionTypes []string
+	ignoreTypesIfMissing           []string
+	stepCounter                    bool
+}
+
+// ApplyOptions applies the given list options on these options,
+// and then returns itself (for convenient chaining).
+func (o *SummaryOptions) ApplyOptions(opts []SummaryOption) *SummaryOptions {
+	for _, opt := range opts {
+		opt.ApplyToSummary(o)
+	}
+	return o
+}
+
+// NewSummaryCondition creates a new condition by summarizing a set of conditions from an object.
+// If any of the condition in scope does not exist in the source object, missing conditions are considered Unknown, reason NotYetReported.
+//
+// Additionally, it is possible to inject custom merge strategies using the WithMergeStrategy option or
+// to add a step counter to the generated message by using the WithStepCounter option.
+func NewSummaryCondition(sourceObj runtime.Object, targetConditionType string, opts ...SummaryOption) (*metav1.Condition, error) {
+	summarizeOpt := &SummaryOptions{
+		mergeStrategy: newDefaultMergeStrategy(),
+	}
+	summarizeOpt.ApplyOptions(opts)
+
+	conditions, err := getConditionsWithOwnerInfo(sourceObj)
+	if err != nil {
+		return nil, err
+	}
+
+	expectedConditionTypes := sets.New[string](summarizeOpt.conditionTypes...)
+	ignoreTypesIfMissing := sets.New[string](summarizeOpt.ignoreTypesIfMissing...)
+	existingConditionTypes := sets.New[string]()
+
+	// Drops all the conditions not in scope for the merge operation
+	conditionsInScope := make([]ConditionWithOwnerInfo, 0, len(expectedConditionTypes))
+	for _, condition := range conditions {
+		if !expectedConditionTypes.Has(condition.Type) {
+			continue
+		}
+		conditionsInScope = append(conditionsInScope, condition)
+		existingConditionTypes.Insert(condition.Type)
+	}
+
+	// Add the expected conditions which do net exists, so we are compliant with K8s guidelines
+	// (all missing conditions should be considered unknown).
```

Comment:
Let's update to also mention ignoreTypesIfMissing

Reply:
Done